linalg/superlu.cpp

```diff
@@ -130,6 +130,9 @@ SuperLURowLocMatrix::SuperLURowLocMatrix( const HypreParMatrix & hypParMat )
    // hypre_CSRMatrix.
    hypre_CSRMatrix * csr_op = hypre_MergeDiagAndOffd(parcsr_op);
    hypre_CSRMatrixSetDataOwner(csr_op,0);
+#if MFEM_HYPRE_VERSION >= 21600
+   hypre_CSRMatrixBigJtoJ(csr_op);
```

Comment:
Is there a way to check if this is going to succeed?  I suppose we already make the assumption in various places that big ints are not necessary.  If so, I wonder how long that assumption will remain valid...

Reply:
In terms of sustainability?

Reply:
I was wondering if HYPRE provided another function that could be called before `hyper_CSRMatrixBigJtoJ` to verify that this function would produce usable results.  All this function does is allocate an integer array and copy data from `big_j` to the new array.  If `big_j` contains very large values this copy will produce invalid integers.  Some sort of error checking should also be done.

Reply:
How about this: we first verify that the number of local columns is small enough to be stored as an `int`.  If not then `big_j` most likely contains column indices that are too large to be stored as `ints` and we issue an error because `hyper_CSRMatrixBigJtoJ` will most likely fail.

Reply:
I like that approach.

Reply:
Hi, @jandrej, I just put in such a check but I don't have a copy of the appropriate version of HYPRE built to test it.  Would you mind testing this change when you get a chance?

Reply:
Seems to work for me.